cvat/apps/engine/models.py

```diff
@@ -535,3 +536,80 @@ class Comment(models.Model):
     message = models.TextField(default='')
     created_date = models.DateTimeField(auto_now_add=True)
     updated_date = models.DateTimeField(auto_now=True)
+
+class CloudProviderChoice(str, Enum):
+    AWS_S3 = 'AWS_S3_BUCKET'
+    AZURE_CONTAINER = 'AZURE_CONTAINER'
+    GOOGLE_DRIVE = 'GOOGLE_DRIVE'
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda x: x.value, cls))
+
+    def __str__(self):
+        return self.value
+
+class CredentialsTypeChoice(str, Enum):
+    # ignore bandit issues because false positives
+    TEMP_KEY_SECRET_KEY_TOKEN_SET = 'TEMP_KEY_SECRET_KEY_TOKEN_SET' # nosec
+    ACCOUNT_NAME_TOKEN_PAIR = 'ACCOUNT_NAME_TOKEN_PAIR' # nosec
+    ANONYMOUS_ACCESS = 'ANONYMOUS_ACCESS'
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda x: x.value, cls))
+
+    def __str__(self):
+        return self.value
+
+class CloudStorage(models.Model):
+    # restrictions:
+    # AWS bucket name, Azure container name - 63
+    # AWS access key id - 20
+    # AWS secret access key - 40
+    # AWS temporary session tocken - None
+    # The size of the security token that AWS STS API operations return is not fixed.
+    # We strongly recommend that you make no assumptions about the maximum size.
+    # The typical token size is less than 4096 bytes, but that can vary.
+    provider_type = models.CharField(max_length=20, choices=CloudProviderChoice.choices())
+    resource = models.CharField(max_length=63)
+    display_name = models.CharField(max_length=63, unique=True)
```

Comment:
@Marishka17  @nmanovic why make this field unique? I think it might be unique in user space, but it should not be globally unique. For example, on Github, I can create a repo named `cvat` in my user space, but on cvat.org I won't be able to create a cloud storage named `aws` if someone has already created it. Can a cloud resources be shared between users?

Reply:
> Can a cloud resources be shared between users?

The created storage will be available to the user who created it and the administrator

> > `specific_attributes` should contain structure like `key1=value1&key2=value2`.
> > In real case e.g `"specific_attributes": "range=eu-west-2"`
> 
> @Marishka17 could you please add a note about that into swagger docs?

possible solutions:

1) Use
```
swagger_schema_fields = {
            'type': openapi.TYPE_OBJECT,
            'title': 'Cloud Storage',
            'properties': {
                'specific_attributes': openapi.Schema(
                    title='Specific attributes',
                    type=openapi.TYPE_STRING,
                    description='structure like key1=value1&key2=value2\n'
                        'supported: range=aws_range',
                ),
            },
            # "required": [...],
         }
```
in CloudStorageSerializer Meta class need to redefine all descriptions that should be generated automatically.

2) move `specific_attributes` into separate class
```
class SpecificAttributes(serializers.Field):
    def to_representation(self, value):
        pass
    def to_internal_value(self, value):
        pass
    class Meta:
        swagger_schema_fields = {
            'type': openapi.TYPE_STRING,
            'title': 'Specific attributes',
            'description': 'structure like key1=value1&key2=value2\n'
                           'supported: range=aws_range',
            'maxLength': '50',
        } 
```
IMHO, it should not be separated into a separate class just because of the addition to the documentation.

3. Use fieldInspector, I chose this solution.